code/game/objects/structures/ladders.dm

```diff
@@ -73,25 +73,29 @@
 	if(busy)
 		to_chat(user, SPAN_WARNING("Someone else is currently using [src]."))
 		return
+
 	var/ladder_dir_name
 	var/obj/structure/ladder/ladder_dest
 	if(up && down)
-		ladder_dir_name = alert("Go up or down the ladder?", "Ladder", "Up", "Down", "Cancel")
-		if(ladder_dir_name == "Cancel")
-			return
-		ladder_dir_name = lowertext(ladder_dir_name)
-		if(ladder_dir_name == "up")
+		var/static/list/direction_selection = list("up" = image(icon = 'icons/mob/radial.dmi', icon_state = "radial_ladder_up"), "down" = image(icon = 'icons/mob/radial.dmi', icon_state = "radial_ladder_down"))
```

Comment:
you're generating the image() every time someone clicks the ladder here no? should only need to be created once